### ISSUE TITLE
Flaresolverr 3.1.2 take to much resource in background

### ISSUE THREAD

**Issue by 3xploiton3** [FlareSolverr/FlareSolverr#772]
### Have you checked our README?

- [X] I have checked the README

### Is there already an issue for your problem?

- [X] I have checked older issues, open and closed

### Have you checked the discussions?

- [X] I have read the Discussions

### Environment

```markdown
- FlareSolverr version: 3.1.2
- Last working FlareSolverr version:
- Operating system: windows 11
- Are you using Docker: [no]
- FlareSolverr User-Agent (see log traces or / endpoint):
- Are you using a VPN: [yes]
- Are you using a Proxy: [no]
- Are you using Captcha Solver: [no]
- If using captcha solver, which one:
- URL to test this issue:
```

### Description

flaresolverr take to much resource in background, even when i closed flaresolverr, chromium still exist

### Logged Error Messages

```text
none
```

### Screenshots

![image](https://user-images.githubusercontent.com/19517680/236110282-56d92bf4-3263-4faf-a3af-43c1ce13f7a5.png)

**Comment by 3xploiton3** [comment]
<details>

![image](https://user-images.githubusercontent.com/19517680/236110619-0deea999-67e1-4ab4-a4d7-4a35d88a6dfe.png)

</details>

**Comment by ilike2burnthing** [comment]
End the Chromium processes, disabled and shut down your AV, run FlareSolverr, use FlareSolverr, see if the issue continues.

**Comment by Noxillio** [comment]
I just encountered the same issue. Left it running overnight and then I wake up and log onto my computer the next morning to find several dozen Chromium processes in the background that were never closed. They too did not go away after closing FlareSolverr.

**Comment by ilike2burnthing** [comment]
See above.

**Comment by Noxillio** [comment]
I only use Windows Defender so I'm not sure what more could possibly be done there. Will use FlareSolverr if I need it again and report again if I encounter the same issue.

**Comment by ilike2burnthing** [comment]
https://github.com/FlareSolverr/FlareSolverr/issues/761#issuecomment-1511004660

**Comment by subik89** [comment]
I have the same issue.
![image](https://user-images.githubusercontent.com/8209905/236452584-1063c606-6be8-4a56-80be-72572c61d438.png)

After few minutes the instances of chrome increased
![image](https://user-images.githubusercontent.com/8209905/236456203-40eb7450-5da4-4d3f-8a4a-2afa5f555ace.png)

I am running the flaresolverr on Windows 11 instance. I have the newest version of google chrome.

**Comment by kolbdog3333** [comment]
I have to agree with this person it does it opens up multiple chromes in background and makes it slower and the chromium browser doesn't exit out  when you close flaresolverr so when you try to use fanficfare or another program normally its slower. Also sometimes it takes 6 tries for it to bypass the captcha page if its for fanfiction.net. Also in task explorer chrome opens instead of chromium

**Comment by BradenWentworth** [comment]
![image](https://user-images.githubusercontent.com/7004039/236632844-40a542bc-2ec0-4d52-a217-e61db2e1f358.png)
Something is really wrong here.

**Comment by ilike2burnthing** [comment]
Again, disable and exit your AV. Close FlareSolverr and any remaining instances of Chromium. Restart and use FlareSolverr.

If that works, restart your AV, and create an exclusion for FlareSolverr and Chromium.

If however the issue continues, then post here with any relevant information. All 'same here' comments will be hidden as spam.

**Comment by kolbdog3333** [comment]
> Again, disable and exit your AV. Close FlareSolverr and any remaining instances of Chromium. Restart and use FlareSolverr.
> 
> If that works, restart your AV, and create an exclusion for FlareSolverr and Chromium.
> 
> If however the issue continues, then post here with any relevant information. All 'same here' comments will be hidden as spam.

I tried that and it doesn't work. It still opens up15 instances of chromium

**Comment by kolbdog3333** [comment]
Also it opens Chrome in mine not chromium

**Comment by ilike2burnthing** [comment]
> then post here with any relevant information

e.g. the equivalent information provided in the OP, what AV you're using, and specifically what steps you took.

**Comment by kolbdog3333** [comment]
I'm using AVG and I have it disabled I only enable it if I'm downloading something. Also I downloaded the release version for Windows since I dont have a linux or a docker.

**Comment by ilike2burnthing** [comment]
Please re-read my comment and provide the information requested.

**Comment by kolbdog3333** [comment]
Those are the steps I took I can't explain it any better.

**Comment by ilike2burnthing** [comment]
> the equivalent information provided in the OP

That does not mean just 'Windows'.

> I'm using AVG and I have it disabled I only enable it if I'm downloading something.

How are you disabling it? Is Windows Defender therefore running instead?

**Comment by kolbdog3333** [comment]
I dont have Windows Defender I have Windows 8.1 it no longer supports it and I disabled AVG by using the settings.

**Comment by ilike2burnthing** [comment]
Windows 8.1 includes Defender.

If you can answer questions fully I might be able to help.

**Comment by kolbdog3333** [comment]
> Windows 8.1 includes Defender.
> 
> If you can answer questions fully I might be able to help.

My Windows Defender no longer works so I know that's not the problem. I only have AVG and MalwareBytes and they are both disabled when I run flaresolverr. And Chromium doesn't show up in Task Manager Chrome does. Even though flaresolverr says it's running Chromium. And 7 or more instances show up and it slows down the pc.

**Comment by kolbdog3333** [comment]
Not sure elae how to describe probelm since that's what happens

**Comment by ilike2burnthing** [comment]
For the last time:

> [post] the equivalent information provided in the OP

Just saying 'Windows 8.1' is not sufficient.

> How are you disabling [AVG]?

Just saying 'using the settings' is not sufficient.

**Comment by kolbdog3333** [comment]
> For the last time:
> 
> > [post] the equivalent information provided in the OP
> 
> Just saying 'Windows 8.1' is not sufficient.
> 
> > How are you disabling [AVG]?
> 
> Just saying 'using the settings' is not sufficient.

What do you mean by OP? I am disabling it by going to settings and disabling it until next restart.

**Comment by kolbdog3333** [comment]
> > Windows 8.1 includes Defender.
> > If you can answer questions fully I might be able to help.
> 
> My Windows Defender no longer works so I know that's not the problem. I only have AVG and MalwareBytes and they are both disabled when I run flaresolverr. And Chromium doesn't show up in Task Manager Chrome does. Even though flaresolverr says it's running Chromium. And 7 or more instances show up and it slows down the pc.

And here is fully issue even with AVG disabled using the settings in the program with it disabked until next system restart..

**Comment by ilike2burnthing** [comment]
Original Post.

What setting? If you feel you can't describe it, take a screenshot.

Are you then shutting down AVG after disabling it?

**Comment by kolbdog3333** [comment]
> Original Post.
> 
> What setting? If you feel you can't describe it, take a screenshot.
> 
> Are you then shutting down AVG after disabling it?

I told you I go into settings in AVG then disable the antivirus until next sytem restart.

**Comment by kolbdog3333** [comment]
Also how would I shut it down. I cant force close it in Task Manager I tried it wont let me.

**Comment by ilike2burnthing** [comment]
Given that AVG has individual 'shields' that you can disable independent of disabling the whole AV, just telling me that you're going into the settings is utterly meaningless.

To save another 8hrs of this waste of time, just boot to safe mode with networking and try using FlareSolverr.

**Comment by kolbdog3333** [comment]
> Given that AVG has individual 'shields' that you can disable independent of disabling the whole AV, just telling me that you're going into the settings is utterly meaningless.
> 
> To save another 8hrs of this waste of time, just boot to safe mode with networking and try using FlareSolverr.

I'm using AVG free not paid. I can only disable it from scanning anything when I go into settings and disable until next restart.

**Comment by kolbdog3333** [comment]
Also it can only scan files there are no shields I have it set up that it will only detect the exe if first opened. And avg never detected it. And it's disabled so it doesnt scan anything.

**Comment by kolbdog3333** [comment]
Does that help

**Comment by kolbdog3333** [comment]
Also I cant run my pc in safe mode since their is no keyboard it only detects the keyboard once Windows logins in.

**Comment by kolbdog3333** [comment]
I cant adjust any settings so cant use safe mode.

**Comment by kolbdog3333** [comment]
Its a touch screen with no keyboard or mouse

**Comment by ilike2burnthing** [comment]
Create exclusions for FlareSolverr and Chromium/Chrome.

Failing that, uninstall AVG.

**Comment by kolbdog3333** [comment]
How can I make it so when I run Flaresolverr on Windows it only runs one hidden browser so multiple ones dont open up each time I try to use flaresolverr I don't want to keep Task Manager open the entire time. Like force close the other chromium or chrome browsers after each run or session.

**Comment by ilike2burnthing** [comment]
Have you tried the above suggestions?

**Comment by kolbdog3333** [comment]
> Have you tried the above suggestions?

Yes but it doesn't solve the issue it just allows it to run without any problems. It doesn't do anything to solve more than one hidden browser

**Comment by ilike2burnthing** [comment]
So has removing the AV from the equation has resolved the zombie processes issue?

Are you now just wanting to run a single chrome process? If so, I would guess you'd need to add `options.add_argument('--single-process')` to `utils.py` and recompile, but the flag `--single-process` only works for Chromium not Chrome (Google's limitation), and currently FlareSolverr on Windows only supports Chrome.

**Comment by ngosang** [comment]
If I make many requests at the same time I can reproduce the issue, but the processes are killed after a while.

**Comment by hwertz** [comment]
I saw the behavior of flaresolverr not closing chrome or chromedriver on Ubuntu (this server I think is running 20.04 still, but I put python 3.11 on there and put flaresolverr in a venv).  As far as I can tell, if flaresolverr gets a chance to do a clean shutdown, it closes chromedriver and chrome; but (while I was setting it up I restarted it quite a few times), sometimes it doesn't.  I just killed chromedriver and chrome manually (since I was already killing flaresolverr manually to restart it at that point anyway) and it was fine.

I found the resource usage to be fine; the installation I set up is only on a 1GB virtual server, so chrome uses up a fair portion of it.. but that's chrome, flaresolverr itself is using very little for me.  (Happily, my application, flaresolverr, and chrome do fit with about 200MB to spare.)

**Comment by dojitr** [comment]
im have the same issue. been using the old version for so long and i had noticed today that it no longer works. so i checked and found that my version was too old so i updated to the latest, it works but i noticed when im playing some files on emby it suddenly stops then i will lose access to the server. found out the flaresolverr has very high cpu usage and a lot of open chromium processes..  so many my cpu usage reaches 100%.

no AV
firewall disabled

still the same

any suggestions for a fix?
thanks!

**Comment by ngosang** [comment]
@dojitr are you using linux? Docker?

**Comment by dojitr** [comment]
> 

hi, sorry forgot to provide some details..
im using it on windows 19

**Comment by HeresJonny** [comment]
Same here, Windows 10 22H2. flaresolverr 3.2.0 sometimes starts 2 or more Chrome processes which hog the CPU and slow down the complete system to a crawl. Defender is deactivated by Group Editor, no other AV present. If the Chrome processes are killed by task manager, they will not start again the next time flaresolverr is started.

**Comment by DrStrange** [comment]
I am also seeing high CPU usage with FlareSolverr v3.2.0. I did not notice this in v3.1.2
After the PC starts, this is the command line in Task Manager for the Chrome instance,titled "Chromium (32bit)":

`"C:\Program Files\chrome\chrome.exe" --type=renderer --user-dat-dir="C\Users\<InsertUser>\AppData\local\Temp\tmp19w6u7v7" --first-renderer-process --no-sandbox --file-url-path-alias="/gen=C\Program Files\Flaresolverr\chrome\gen" --log-level=0  --no-zygote --remote-debugging-port=1113 --test-type --video-capture-use-gpu-memory-buffer --lang=en-GB --device-scale-factor=1.75 --num-raster-threads=4 --enable-main-frame-before-activation --renderer-client-id=6 --time-ticks-at-unix-epoch=-1686354149219930 --launch-time-ticks=46976420 --mojo-platform-channel-handle=3216 --field-trial-handle=2096,i,9394694510220931050,17729409452463252051,131072 /prefetch:1`
CPU usage is around 12%

After around five minutes another instance of Chrome starts, also entitled "Chromium (32bit)" also using around 12% of CPU. This is the command line:
`"C:\Program Files\chrome\chrome.exe" --type=renderer --user-dat-dir="C\Users\<InsertUser>\AppData\local\Temp\tmpyoi51rx2" --no-sandbox --file-url-path-alias="/gen=C\Program Files\Flaresolverr\chrome\gen" --log-level=0  --no-zygote --remote-debugging-port=1300 --test-type --video-capture-use-gpu-memory-buffer --lang=en-GB --device-scale-factor=1.75 --num-raster-threads=4 --enable-main-frame-before-activation --renderer-client-id=7 --time-ticks-at-unix-epoch=-1686354149219930 --launch-time-ticks=1128223677 --mojo-platform-channel-handle=4136 --field-trial-handle=1964,i,580597598255223205,17606172434374018863,131072 /prefetch:1`

tmp19w6u7v7 and tmpyoi51rx2 do not exist in AppData\local\Temp
In the command line I have replaced my user name with <InsertUser>

I do not use Docker/VPN/Proxy/Captcha Solver.
OS: Windows 10 22H2 64 bit v10.0.19045.3031
FlareSolverr UserAgent: "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36"

I only use Windows Defender for an AV. Disabling Windows Defender then rebooting the PC does not prevent the chrome instances appearing in Task Manager again. I then excluded flaresolverr.exe and chrome.exe in Windows Defender. This also does not stop the instances from appearing in Task Manager after a reboot.
After ending the Chrome instances in Task Manager, flaresolverr.exe continues to run as normal. So far (after waiting 10 minutes) no instance of Chrome has reappeared.

FlareSolverr is set to automatically start at bootup using the following VBscript file in C:\ProgramData\Microsoft\Windows\Start Menu\Programs\StartUp
```
Set WshShell = CreateObject("WScript.Shell") 
WshShell.Run chr(34) & "C:\Program Files\FlareSolverr\Flaresolverr.exe" & Chr(34), 0
Set WshShell = Nothing
```
